### PR TITLE
[node] Allow util.format(), util.formatWithOptions() with no content arguments

### PR DESCRIPTION
**Pull request by eritbh**
This PR updates signatures for Node's `util.format` and `util.formatWithOptions` to accept no content arguments (the latter still has a required options argument). This case is not explicitly mentioned in the [Node docs](https://nodejs.org/api/util.html#util_util_format_format_args), but in all supported versions of Node, this results in an empty string:

```
Welcome to Node.js v14.15.1.
Type ".help" for more information.
> util.format()
''
> util.formatWithOptions({})
''
```

My use case for this is to pass arrays of unknown length into `util.format` for display as part of a logging utility, e.g.:

```ts
function formatArgs (...args: any[]): string {
    return util.format(...args);
}
```

I've also added several test cases for `util.format`, since none existed before. Apologies for the force pushes, I accidentally committed unrelated whitespace changes and didn't check carefully enough before submitting the PR.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
  - I was unable to run prettier without touching many unrelated files, so I haven't run it. Would another PR to run prettier on the node directory be accepted?
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/util.html#util_util_format_format_args
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by Geo1088** [comment]
CI failed on unrelated module. `npm test node` succeeds on my machine.

File: types/node/util.d.ts [hunk 1]
```diff
@@ -6,7 +6,9 @@ declare module "util" {
         stylize(text: string, styleType: Style): string;
     }
     function format(format: any, ...param: any[]): string;
+    function format(...param: any[]): string;
     function formatWithOptions(inspectOptions: InspectOptions, format: string, ...param: any[]): string;
+    function formatWithOptions(inspectOptions: InspectOptions, ...param: any[]): string;
```
Comment (ExE-Boss) [inline types/node/util.d.ts:11]:
```suggestion
    function format(format?: any, ...param: any[]): string;
    function formatWithOptions(inspectOptions: InspectOptions, format?: any, ...param: any[]): string;
```

File: types/node/v11/util.d.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 declare module "util" {
     interface InspectOptions extends NodeJS.InspectOptions { }
     function format(format: any, ...param: any[]): string;
+    function format(...param: any[]): string;
     function formatWithOptions(inspectOptions: InspectOptions, format: string, ...param: any[]): string;
+    function formatWithOptions(inspectOptions: InspectOptions, ...param: any[]): string;
```
Comment (ExE-Boss) [inline types/node/v11/util.d.ts:6]:
```suggestion
    function format(format?: any, ...param: any[]): string;
    function formatWithOptions(inspectOptions: InspectOptions, format?: any, ...param: any[]): string;
```

File: types/node/v12/util.d.ts [hunk 3]
```diff
@@ -1,7 +1,9 @@
 declare module "util" {
     interface InspectOptions extends NodeJS.InspectOptions { }
     function format(format: any, ...param: any[]): string;
+    function format(...param: any[]): string;
     function formatWithOptions(inspectOptions: InspectOptions, format: string, ...param: any[]): string;
+    function formatWithOptions(inspectOptions: InspectOptions, ...param: any[]): string;
```
Comment (ExE-Boss) [inline types/node/v12/util.d.ts:6]:
```suggestion
    function format(format?: any, ...param: any[]): string;
    function formatWithOptions(inspectOptions: InspectOptions, format?: any, ...param: any[]): string;
```

File: types/node/v13/util.d.ts [hunk 4]
```diff
@@ -6,7 +6,9 @@ declare module "util" {
         stylize(text: string, styleType: Style): string;
     }
     function format(format: any, ...param: any[]): string;
+    function format(...param: any[]): string;
     function formatWithOptions(inspectOptions: InspectOptions, format: string, ...param: any[]): string;
+    function formatWithOptions(inspectOptions: InspectOptions, ...param: any[]): string;
```
Comment (ExE-Boss) [inline types/node/v13/util.d.ts:11]:
```suggestion
    function format(format?: any, ...param: any[]): string;
    function formatWithOptions(inspectOptions: InspectOptions, format?: any, ...param: any[]): string;
```

**Review by ExE-Boss — COMMENTED** [review]
This can just use an optional parameter.

**Comment by Geo1088** [comment]
> This can just use an optional parameter.

Never knew optionals could come before rest parameters, neat. Thanks!

Re-ran `npm test node`, still passing.

**Review by ExE-Boss — APPROVED** [review]

**Review by a-tarasyuk — APPROVED** [review]

**Review by r3nya — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

**Comment by Geo1088** [comment]
Finally got CI to pass, no new changes except pulls from master.

**Review by Flarna — APPROVED** [review]

**Review by ExE-Boss — APPROVED** [review]

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by Geo1088** [comment]
Ready to merge

**Pull request merged**